tsdate/util.py

```diff
@@ -300,3 +300,43 @@ def add_sampledata_times(samples, sites_time):
     copy.sites_time[:] = sites_time
     copy.finalise()
     return copy
+
+
+def change_time_measure(time_ago, breakpoints, time_measure):
+    """
+    Rescales time given a piecewise-constant time measure. To convert from
+    generations to coalescent units, the time measure per epoch should be 2 *
+    effective population size.  To convert from coalescent units to
+    generations, the time measure should be the coalescent rate ``1/(2 * Ne)``.
+
+    :param np.ndarray time_ago: An increasing vector of time points
+    :param np.ndarray breakpoints: Start times of pieces
+    :param np.ndarray time_measure: Time measure within pieces
+
+    :return: Inputs in new time measure
+    """
+
+    # TODO: Make this work for arbitrary positive fuctions? Could use
+    # scipy.integrate.quad, but would need to check for convergence;
+    # as it can fail when function is not sufficiently smooth.
+
+    assert sorted(breakpoints)
```

Comment:
This doesn't assert that they are sorted, it sorts the array and then asserts the array is not empty.

Reply:
I think you want `assert np.all_equal(breakpoints, sorted(breakpoints))`. Or maybe `assert np.all(np.diff(breakpoints) > 0)`?